Sources/MultipartKit/FormDataDecoder/FormDataDecoder+SingleValueContainer.swift

```diff
@@ -0,0 +1,63 @@
+#if canImport(FoundationEssentials)
+    import FoundationEssentials
+#else
+    import Foundation
+#endif
+
+extension FormDataDecoder.Decoder: SingleValueDecodingContainer {
+    func decodeNil() -> Bool {
+        false
+    }
+
+    func decode<T: Decodable>(_: T.Type = T.self) throws -> T {
+        guard let part = data.part else {
+            guard previousCodingPath?.count != codingPath.count || previousType != T.self else {
+                throw DecodingError.dataCorrupted(.init(codingPath: codingPath, debugDescription: "Decoding caught in recursion loop"))
+            }
+
+            return try T(
+                from: FormDataDecoder.Decoder(
+                    codingPath: codingPath, data: data, userInfo: sendableUserInfo, previousCodingPath: codingPath, previousType: T.self
+                )
+            )
+        }
+
+        let decoded =
+            switch T.self {
+            case is MultipartPart<Body>.Type:
+                part as? T
+            case is String.Type:
+                String(bytes: part.body, encoding: .utf8) as? T
+            case let IntType as any FixedWidthInteger.Type:
+                String(bytes: part.body, encoding: .utf8).flatMap(IntType.init) as? T
+            case is Float.Type:
+                String(bytes: part.body, encoding: .utf8).flatMap(Float.init) as? T
+            case is Double.Type:
+                String(bytes: part.body, encoding: .utf8).flatMap(Double.init) as? T
+            case is Bool.Type:
+                String(bytes: part.body, encoding: .utf8).flatMap(Bool.init) as? T
+            case is Data.Type:
+                Data(part.body) as? T
+            case is URL.Type:
+                String(bytes: part.body, encoding: .utf8).flatMap(URL.init(string:)) as? T
```

Comment:
```suggestion
                String(bytes: part.body, encoding: .utf8) as? T
            case let IntType as any FixedWidthInteger.Type:
                String(decoding: part.body, as: UTF8.self).flatMap(IntType.init) as? T
            case is Float.Type:
                String(decoding: part.body, as: UTF8.self).flatMap(Float.init) as? T
            case is Double.Type:
                String(decoding: part.body, as: UTF8.self).flatMap(Double.init) as? T
            case is Bool.Type:
                String(decoding: part.body, as: UTF8.self).flatMap(Bool.init) as? T
            case is Data.Type:
                Data(part.body) as? T
            case is URL.Type:
                String(decoding: part.body, as: UTF8.self).flatMap(URL.init(string:)) as? T
```
Also, for the "basic" types (`String`, integers, `Float`/`Double`, `Bool`) you get a significant speedup by implementing the individual non-generic `decode()` methods for those types rather than doing a runtime type check in the generic method.